src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -636,8 +636,8 @@ namespace AppInstaller::CLI::Workflow
                 changes.size(),
                 findByManifest.Matches.size(),
                 packagesInBoth.size(),
-                toLog ? static_cast<std::string_view>(toLog->GetProperty(PackageVersionProperty::Name)) : "",
-                toLog ? static_cast<std::string_view>(toLog->GetProperty(PackageVersionProperty::Version)) : "",
+                toLog ? static_cast<std::string>(toLog->GetProperty(PackageVersionProperty::Name)) : "",

```

Comment:
Why here but not the two lines below?  The difference is rvalue vs lvalue, but the lifetime of the rvalue is until the end of the statement.  So converting to string_view is just as valid.  This is either a mistake/overcaution in the analysis, or an error in my understanding.

Reply:
The code analysis error is C26449( [https://docs.microsoft.com/en-us/cpp/code-quality/c26449?view=msvc-160](https://docs.microsoft.com/en-us/cpp/code-quality/c26449?view=msvc-160) )
It specifically checks view created from temporary. It is most likely an overcaution because we immediately use the view within the statement. This is to make code analysis happy.

---
In reply to: [613644294](https://github.com/microsoft/winget-cli/pull/856#discussion_r613644294) [](ancestors = 613644294)

Reply:
I think the ternary is confusing it, since that link specifically calls out:
```
Temporaries created for function call arguments are not flagged. It is safe to pass spans from such temporaries if target functions don’t retain data pointers in external variables.
```